client/question/services/RunestoneHandlerService.js

```diff
@@ -0,0 +1,31 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service that handles all TIE interactions with Runestone.
+ */
+tie.factory('RunestoneHandlerService', ['$window', function($window) {
+  return {
+    /**
+     * Sends the message to the Runestone parent frame via postMessage.
+     *
+     * @param {String} messageType Type of message that is being sent.
+     * @param {String} messageData Data to be sent to parent frame.
+     */
+    sendMessage: function(messageType, messageData) {
+      $window.parent.postMessage(
+        JSON.stringify(messageData), 'https://runestone.academy');
```

Comment:
Pull target out into config constant. Set to null for standalone version. Set it to runestone in server version. Do "if" check here to determine if constant is set before making any postMessage calls.

Reply:
Done.